src/bot/commands/stocks.js

```diff
@@ -1,44 +1,138 @@
+  return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+}
+
+const API = 'https://cloud.iexapis.com/stable/stock/';
+
+const TOKEN = `quote?token=${process.env.IEXCLOUD_KEY}`;
 
-async function getStock(sign) {
-  return Promise.resolve(sign);
+function createStocksEmbed(data) {
+  const change = `Change ${
+    data.change >= 0
+      ? ':chart_with_upwards_trend:'
+      : ':chart_with_downwards_trend:'
+  }`;
+  const embed = new RichEmbed({
+    description: `*Stock Symbol: $${data.symbol}*`,
+    color: 16318579,
+    timestamp: `${fromUnixTime(data.latestUpdate / 1000)}`,
+    footer: {
+      text: 'Data provided by IEX Cloud',
+    },
+    author: {
+      name: `${data.companyName} (Click here for more info!)`,
+      url: `https://finance.yahoo.com/quote/${data.symbol}/`,
+    },
+    fields: [
+      {
+        name: 'Open :bell:',
+        value: `${data.open}`,
+        inline: true,
+      },
+      {
+        name: 'Closing :no_bell:',
+        value: `${data.close}`,
+        inline: true,
+      },
+      {
+        name: 'High :dollar:',
+        value: `${data.high}`,
+        inline: true,
+      },
+      {
+        name: 'Low :fire:',
+        value: `${data.low}`,
+        inline: true,
+      },
+      {
+        name: `${change}`,
+        value: `${data.change}`,
+        inline: true,
+      },
+      {
+        name: '52 Week High  :money_with_wings:',
+        value: `${data.week52High}`,
+        inline: true,
+      },
+      {
+        name: '52 Week Low :scream:',
+        value: `${data.week52Low}`,
+        inline: true,
+      },
+      {
+        name: 'Market Cap :moneybag:',
+<<<<<<< HEAD
+        value: `$${data.marketCap.toLocaleString()}`,
+=======
+        value: `$${numberWithCommas(data.marketCap)}`,
+>>>>>>> 6c7c0f56db1965a20bdff7fc454b184a98446208
+        inline: true,
+      },
+    ],
+  });
+  return embed;
 }
 
 /** @type {import('../../typedefs').Command} */
 export default async ({ aquarius, analytics }) => {
-  aquarius.onCommand(
-    /^stocks(?: (?<duration>(?:1d|1w|1m|3m|6m|1y|2y|5y|10y)))? (?<sign>.+)$/i,
-    async (message, { groups }) => {
-      const data = await getStock(groups.sign);
-
-      /**
-       * Chart
-       * Open
-       * High
-       * Low
-       * Vol
-       * P/E
-       * Market Cap
-       * 52w High
-       * 52w Low
-       * Avg Vol
-       * Yield
-       * Beta
-       * EPS
-       */
-
-      const embed = new RichEmbed().setURL(data.url);
+  aquarius.onCommand(/^stocks (?<sign>.+)$/i, async (message, { groups }) => {
+    const check = aquarius.permissions.check(
+      message.guild,
+      ...info.permissions
+    );
 
+    if (!check.valid) {
+      log('Invalid permissions');
+      message.channel.send(
+        aquarius.permissions.getRequestMessage(check.missing)
+      );
+      return;
+    }
+
+    try {
+      aquarius.loading.start(message.channel);
+      const response = await fetch(`${API}/${groups.sign}/${TOKEN}`);
```

Comment:
Can we just use the raw string here instead of variables? Don't think they're reused, right?